### DISCUSSION TITLE
How does wails achieve real-time communication between the front and back ends

### DISCUSSION THREAD

**Discussion by wenjian022** [wailsapp/wails#2844] [Q&A] [answered]
Please, can I implement real-time communication with the frontend in the Golang backend of Walis, I need to do a real-time log view and a progress bar

**Comment by NanoNik** [comment, marked as answer]
You can use [events](https://wails.io/docs/reference/runtime/events) from wails runtime to send logs and progress bar value to your frontend.

Or you can start local http server and use [SSE](https://en.wikipedia.org/wiki/Server-sent_events)

**Reply by wenjian022** [reply]
Thanks, I've implemented that via events.

**Reply by Infinage** [reply]
Apologies for touching on a very old thread, is this still the case with wailsV3?

I've noticed that sse events works fine on linux but the events are never flushed on windows. I could not find any official documentation on why sse wouldn't work here
